### ISSUE TITLE
Training Multi-band MelGAN gets huge loss value and cannot recover

### ISSUE THREAD

**Issue by ttsking** [TensorSpeech/TensorFlowTTS#647]
I use a customized Baker Dataset to train the Multi-band MelGAN model. It's fine at the beginning and the lost value suddenly goes huge.

I tested training from scratch:
![tensorboard_from scratch](https://user-images.githubusercontent.com/74232385/129327438-0c05569a-f843-4170-b66d-3b0fb63b015e.jpeg)

and Fine-tune:
![tensorboard_finetune](https://user-images.githubusercontent.com/74232385/129327437-0439797e-d895-46e9-b534-b58099d35288.jpeg)

If i set --generator_mixed_precision 1, the loss value become NaN. If i set  --generator_mixed_precision 0, i can get the loss value, but it is larger than 3000:

```
2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_adversarial_loss = 0.0000.
2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_subband_spectral_convergence_loss = 7186.2988.
2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_subband_log_magnitude_loss = 2.5740.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_fullband_spectral_convergence_loss = 52.8283.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_fullband_log_magnitude_loss = 2.9583.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_gen_loss = 3622.3291.
2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_real_loss = 0.0000.
2021-08-13 08:13:47,477 (base_trainer:657) INFO: (Step: 132200) train_fake_loss = 0.0000.
2021-08-13 08:13:47,477 (base_trainer:657) INFO: (Step: 132200) train_dis_loss = 0.0000.
```

**Comment by ttsking** [comment]
Duplicated post for the poor network. Close this.

**Comment by TheHonestBob** [comment]
> I use a customized Baker Dataset to train the Multi-band MelGAN model. It's fine at the beginning and the lost value suddenly goes huge.
> 
> I tested training from scratch: ![tensorboard_from scratch](https://user-images.githubusercontent.com/74232385/129327438-0c05569a-f843-4170-b66d-3b0fb63b015e.jpeg)
> 
> and Fine-tune: ![tensorboard_finetune](https://user-images.githubusercontent.com/74232385/129327437-0439797e-d895-46e9-b534-b58099d35288.jpeg)
> 
> If i set --generator_mixed_precision 1, the loss value become NaN. If i set --generator_mixed_precision 0, i can get the loss value, but it is larger than 3000:
> 
> ```
> 2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_adversarial_loss = 0.0000.
> 2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_subband_spectral_convergence_loss = 7186.2988.
> 2021-08-13 08:13:47,475 (base_trainer:657) INFO: (Step: 132200) train_subband_log_magnitude_loss = 2.5740.
> 2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_fullband_spectral_convergence_loss = 52.8283.
> 2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_fullband_log_magnitude_loss = 2.9583.
> 2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_gen_loss = 3622.3291.
> 2021-08-13 08:13:47,476 (base_trainer:657) INFO: (Step: 132200) train_real_loss = 0.0000.
> 2021-08-13 08:13:47,477 (base_trainer:657) INFO: (Step: 132200) train_fake_loss = 0.0000.
> 2021-08-13 08:13:47,477 (base_trainer:657) INFO: (Step: 132200) train_dis_loss = 0.0000.
> ```

hi，I get the same problem，have you solved it?

**Comment by l525889198** [comment]
I have the same problem 。
Baker Dataset to train the Multi-band MelGAN model.
 It's fine at the beginning and the lost value suddenly goes huge， at 130K，then，
eval_subband_spectral_convergence_loss = nan.
eval_subband_log_magnitude_loss = nan.
eval_fullband_spectral_convergence_loss = nan.
eval_fullband_log_magnitude_loss = nan.

**Comment by TheHonestBob** [comment]
> I have the same problem 。 Baker Dataset to train the Multi-band MelGAN model. It's fine at the beginning and the lost value suddenly goes huge， at 130K，then， eval_subband_spectral_convergence_loss = nan. eval_subband_log_magnitude_loss = nan. eval_fullband_spectral_convergence_loss = nan. eval_fullband_log_magnitude_loss = nan.

baker data has no problem,if i use my data will have this problem, when you create dataset , you can set batch_max_steps=1s in  def collater func

**Comment by l525889198** [comment]
> > I have the same problem 。 Baker Dataset to train the Multi-band MelGAN model. It's fine at the beginning and the lost value suddenly goes huge， at 130K，then， eval_subband_spectral_convergence_loss = nan. eval_subband_log_magnitude_loss = nan. eval_fullband_spectral_convergence_loss = nan. eval_fullband_log_magnitude_loss = nan.
> 
> baker data has no problem,if i use my data will have this problem, when you create dataset , you can set batch_max_steps=1s in def collater func

3Q，I will try it
